ProcessOptimizer/model_systems/branin_hoo.py

```diff
@@ -1,20 +1,61 @@
 import numpy as np
-from ProcessOptimizer.model_systems import ModelSystem
-from ProcessOptimizer.model_systems.benchmarks import branin
-from ProcessOptimizer.space import Real
 
-# Take the score function from the benchmarks file
-# Add noise
-def score(x, rng=np.random.default_rng(), noise_std=0.02): 
-    return branin(x) + rng.normal(scale=noise_std)
+from .model_system import ModelSystem
+from ..space import Real
 
-# Define the relevant parameter space
-space = []
-space += [Real(-5,10,name='x1')]
-space += [Real(0,15,name='x2')]
 
-# Set the true minimum value
-true_min = 0.397887
+# Defining the Branin-Hoo benchmark function
+
+
+def branin_score(x):
+    """
+    The Branin-Hoo function.
+
+    Intended parameter Space during benchmark use:
+        [(-5.0, 10.0), (0.0, 15.0)]
+        If x contains more than two dimensions the excess are ignored.
+
+    Global minimum value, f(x*): 0.3979
+    Global mimimum location, x*: (-pi, 12.275), (+pi, 2.275), and (9.425, 2.475)
+    Local minima locations, x**: None.
+    Global maximum value, f(x+): 308.13
+    Global maximum location, x+: (-5.0, 0.0)
+
+    More details: <http://www.sfu.ca/~ssurjano/branin.html>
+
+    Parameters
+    ----------
+    * 'x' [array of floats of length >=2]:
+        The point to evaluate the function at.
+
+    Returns
+    -------
+    * 'score' [float]:
+        The score of the system at x.
+    """
+    # Define the constants that are canonically used with this function.
+    a = 1
+    b = 5.1 / (4 * np.pi**2)
+    c = 5.0 / np.pi
+    r = 6
+    s = 10
+    t = 1.0 / (8 * np.pi)
+    return (
+        a * (x[1] - b * x[0] ** 2 + c * x[0] - r) ** 2 + s * (1 - t) * np.cos(x[0]) + s
+    )
+
 
 # Create model system
-branin_hoo = ModelSystem(score, space, true_min)
\ No newline at end of file
+branin = ModelSystem(
+    branin_score,
+    [Real(-5, 10, name="x1"), Real(0, 15, name="x2")],
+    noise_model={"model_type": "constant", "noise_size": 0.02},
```

Comment:
We should probably go through all of the noise levels, I don't think I have given them enough thought.

And this is almost certainly too small.

Reply:
Made proportional to the signal, at the 10% level. This might seem high compared to the 1% default for constant noise, but the 1% is relative to the entire span, which mostly means the worst value, while the 1% is relative to the signal at the sampling point, which is much lower.